src/bin/download_sysext.rs

```diff
@@ -342,6 +347,58 @@ async fn main() -> Result<(), Box<dyn Error>> {
     fs::create_dir_all(&unverified_dir)?;
     fs::create_dir_all(&temp_dir)?;
 
+    // The default policy of reqwest Client supports max 10 attempts on HTTP redirect.
+    let client = Client::builder().redirect(Policy::default()).build()?;
+
+    // If input_xml exists, simply read it.
+    // If not, try to read from payload_url.
+    let res_local = match args.input_xml {
+        Some(name) => {
+            if name == "-".to_string() {
+                Some(io::read_to_string(io::stdin())?)
+            } else {
+                let file = File::open(name)?;
+                Some(io::read_to_string(file)?)
+            }
+        }
+        None => None,
+    };
+
+    let response_text = match res_local {
```

Comment:
Let's remove the `let  ... =` part.

Reply:
And best would be to do a `match (res_local, args.payload_url)` for more clarity of the cases.

Reply:
Changed to match (tuple).

I had to keep `let response_text`, which is reused below.